models/issues/issue.go

```diff
@@ -1464,6 +1469,15 @@ func applyReviewRequestedCondition(sess *xorm.Session, reviewRequestedID int64)
 			reviewRequestedID, ReviewTypeApprove, ReviewTypeReject, ReviewTypeRequest, reviewRequestedID)
 }
 
+func applySubscribedCondition(sess *xorm.Session, subscriberID int64) *xorm.Session {
+	return sess.And("issue.id NOT IN (SELECT issue_id FROM issue_watch WHERE is_watching = ? AND user_id = ?)", false, subscriberID).
```

Comment:
can we convert it into xorm.builder ?

Reply:
```go
builder.NotIn("issue.id", builder.
	Select("issue_id").
	From("issue_watch").
	Where(builder.Eq{"is_watching": false, "user_id": subscriberID}),
).And(
	builder.Or(
		builder.In("issue.id", builder.
			Select("issue_id").
			From("issue_watch").
			Where(builder.Eq{"is_watching": true, "user_id": subscriberID}),
		),
		builder.In("issue.id", builder.
			Select("issue_id").
			From("comment").
			Where(builder.Eq{"poster_id": subscriberID}),
		),
		builder.Eq{"issue.poster_id": subscriberID},
		builder.In("issue.repo_id", builder.
			Select("id").
			From("watch").
			Where(builder.Eq{"user_id": subscriberID, "mode": true}),
		),
	),
)
```
But you have to check which condition should be where. At the moment the original code has `(a AND b) OR c OR d OR e`. My code generates `a AND (b OR c OR d OR e)` which is more like your answer but not what's in the original code. But it demonstrates how you can group expressions with xorm.